discovery.md

```diff
@@ -199,31 +202,68 @@ with the wire format/API defined by this specification.
 The following sections define the attributes that appear in a Service
 entity.
 
-##### name
+##### id
 - Type: `String`
 - Description: A unique identifier for this Service. This value MUST be
-  unique within the scope of this Discovery Endpoint.
+  globally unique. While other metadata about this Service MAY change,
+  this value MUST NOT so that clients can use this attribute to know whether
+  a Service returned by a query is the same Service returned by a previous
+  query.
+
+  Typically, this value is defined by the Discovery Endpoint, or one of the
+  components behind it. However, there might be cases where the value is
+  provided to the Discovery Endpoint, for example, during an "import" type of
+  operation.
+
+  Whether a change to a Service would result in changing of the Service's
+  metadata (except `id`) and thus be just an update of an existing Service,
+  or whether the change would result in a brand new Service (with a new `id`)
+  is not defined by this specification.
+
+  For example, if a Service's implementation is upgraded to a new version
+  then whether this would result in a new Service (and `id`) or is an update
+  to the existing Service's metadata, is an implementation choice. Likewise,
+  this specifcation makes no statement, or guarantees, as to the forwards or
+  backwards compatibility a Service as it changes over time.
+
+  Note, unlike the `name` attribute which only needs to be unique within the
+  scope of a single Discovery Endpoint, the global uniqueness aspect of this
+  attribute allows for the discovery of the same Service across multiple
+  Discovery Endpoints.
+
+  See the Primer for more information.
+- Constraints:
+  - REQUIRED
+  - MUST be a valid UUID per RFC4122.
+- Examples:
+  - `bf5ff5cc-d059-4c79-a89a-2513e45a1340`
+
+##### name
+- Type: `String`
+- Description: A unique human readable identifier for this Service. This
+  value MUST be unique (case insensitive) within the scope of this Discovery
+  Endpoint. Note, this differs from the `id` attribute which is globally
+  unique.
 - Constraints:
   - REQUIRED
-  - MUST be a valid `fsegment` per RFC1738.
 - Examples:
-  - `storage`
-  - `github`
+  - `my storage service`
+  - `cool git offering`
 
 ##### url
 
 - Type: `URL`
-- Description: A URL that references this Service. This value MUST
+- Description: An absolute URL that references this Service. This value MUST
   be usable in subsequent requests, by authorized clients, to retrieve this
   Service entity.
 - Constraints:
   - REQUIRED
   - MUST be a non-empty URL
-  - MUST end with `fsegments` (per RFC1738) of: `/services/{name}` where
-    `name` is the Service's `name` attribute.
+  - MUST end with `fsegments` (per RFC1738) of: `/services/{id}` where
+    `id` is the Service's `id` attribute.
 - Examples:
-  - `http://example.com/services/storage`
-  - `http://discovery.github.com/services/github`
+  - `http://example.com/services/bf5ff5cc-d059-4c79-a89a-2513e45a1340`
```

Comment:
While I like the idea of having a unique id per service, it doesn't make sense to put the id in the URL in my opinion. The service name needs to be unique per discovery endpoint, so the name can be used in the URL (which of course limits the name to a single word).
If the name changes, the new URL can be retrieved through the discovery API (..../services/{id}).

Reply:
My main motivation for putting the ID into the URL instead of the NAME is because if the NAME changes I wanted a static URL that a client could use w/o worrying about it changing over time. Often these URLs appear as refs in data and asking someone to find/change all possible refs could be quite costly (and racy).

I did think about allowing both ../services/ID and ../services/NAME, but then you run into the fun issue of people using UUID as NAMES and getting conflicts :-)    But we could say "be smart"... dunno